Classes/Command/BaseCommand.php

```diff
@@ -81,9 +82,12 @@ protected function initializeRepositories($storagePid)
             $frameworkConfiguration['persistence']['storagePid'] = MathUtility::forceIntegerInRange((int) $storagePid, 0);
             $configurationManager->setConfiguration($frameworkConfiguration);
 
-            $this->collectionRepository = GeneralUtility::makeInstance(CollectionRepository::class);
-            $this->documentRepository = GeneralUtility::makeInstance(DocumentRepository::class);
-            $this->libraryRepository = GeneralUtility::makeInstance(LibraryRepository::class);
+            // TODO: When we drop support for TYPO3@9, we needn't/shouldn't use ObjectManager anymore
+            $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+
```

Comment:
AFAIK, that's needed in TYPO3 v9 (in my v9 setup, it didn't work before). The reason is that the repositories take an `ObjectManager` in their constructor, which is automatically injected via `makeInstance` only from v10 on. A variant of this might be to pass `objectManager` to `makeInstance`.

For reference, v10 has some DI code here:
https://github.com/TYPO3/typo3/blob/10.4/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L3471-L3477

It's not present in the v9 version:
https://github.com/TYPO3/typo3/blob/9.5/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L3690

Reply:
Ah, ok. I'll must check with TYPO3 9 again. Currently my development system is TYPO3 10.